Source/EasyNetQ/PersistentConnection.cs

```diff
@@ -1,195 +1,111 @@
+            connection.Dispose();
         }
 
-        private void TryToConnect()
+        private IConnection InitializeConnection()
         {
-            logger.Debug("Trying to connect");
-            if (disposed) return;
-
-            connectionFactory.Reset();
-            do
-            {
-                try
-                {
-                    connection = connectionFactory.CreateConnection(); // A possible dispose race condition exists, whereby the Dispose() method may run while this loop is waiting on connectionFactory.CreateConnection() returning a connection.  In that case, a connection could be created and assigned to the connection variable, without it ever being later disposed, leading to app hang on shutdown.  The following if clause guards against this condition and ensures such connections are always disposed.
-                    if (disposed)
-                    {
-                        connection.Dispose();
-                        break;
-                    }
-
-                    try
-                    {
-                        OnConnected();
-                    }
-                    catch
-                    {
-                        connection.Dispose();
-                        throw;
-                    }
-                    
-                    connection.ConnectionShutdown += OnConnectionShutdown;
-                    connection.ConnectionBlocked += OnConnectionBlocked;
-                    connection.ConnectionUnblocked += OnConnectionUnblocked;
-
-                    logger.InfoFormat(
-                        "Connected to broker {broker}, port {port}, vhost {vhost}",
-                        connectionFactory.CurrentHost.Host,
-                        connectionFactory.CurrentHost.Port,
-                        connectionFactory.Configuration.VirtualHost
-                    );
-
-                    connectionFactory.Success();
-                }
-                catch (Exception ex) when (ex is SocketException || ex is BrokerUnreachableException || ex is TimeoutException)
-                {
-                    LogException(ex);
-                }
-            } while (!disposed && connectionFactory.Next());
-
-            if (!connectionFactory.Succeeded && !disposed)
+            var endpoints = connectionConfiguration.Hosts.Select(x =>
             {
-                logger.ErrorFormat("Failed to connect to any Broker. Retrying in {connectInterval}", connectionFactory.Configuration.ConnectIntervalAttempt);
-                StartTryToConnect();
-            }
+                var endpoint = new AmqpTcpEndpoint(x.Host, x.Port);
+                if (x.Ssl.Enabled)
+                    endpoint.Ssl = x.Ssl;
+                else if (connectionConfiguration.Ssl.Enabled)
+                    endpoint.Ssl = connectionConfiguration.Ssl;
+                return endpoint;
+            }).ToArray();
+            var newConnection = connectionFactory.CreateConnection(endpoints);
```

Comment:
This line will thrown an exception when the broker is down, this completely change our connection management.